sys-devel/binutils/binutils-2.41.recipe

```diff
@@ -0,0 +1,184 @@
+- ar - a utility for creating, modifying and extracting of archives.
+- c++filt - filter to demangle encoded C++ symbols.
+- gold - a linker for ELF files.
+- nm - lists symbols from object files.
+- objcopy - copys and translates object files.
+- objdump - displays information from object files.
+- ranlib - generates an index to the contents of an archive.
+- readelf - displays information from any ELF format object file.
+- size - lists the section sizes of an object or archive file.
+- strings - lists printable strings from files.
+- strip - discards symbols."
+HOMEPAGE="https://www.gnu.org/software/binutils/"
+COPYRIGHT="1988-2023 Free Software Foundation, Inc."
+LICENSE="GNU GPL v3
+	GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://ftp.gnu.org/gnu/binutils/binutils-$portVersion.tar.xz"
+SOURCE_DIR="binutils-$portVersion"
+CHECKSUM_SHA256="ae9a5789e23459e59606e6714723f2d3ffc31c03174191ef0d015bdf06007450"
+SOURCE_FILENAME="binutils-$portVersion.tar.xz"
+
+ARCHITECTURES="?all"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	binutils$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:addr2line$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:ar$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:as$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:c++filt$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:elfedit$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:gprof$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:ld$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:ld.bfd$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:nm$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:objcopy$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:objdump$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:ranlib$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:readelf$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:size$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:strings$secondaryArchSuffix = $portVersion compat >= $portVersion
+	cmd:strip$secondaryArchSuffix = $portVersion compat >= $portVersion
+	lib:libbfd_$binutilsVersion$secondaryArchSuffix = $portVersion compat >= $portVersion
+	lib:libopcodes_$binutilsVersion$secondaryArchSuffix = $portVersion compat >= $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:find
+	cmd:xargs
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:awk
+	cmd:bison
+	cmd:find
+	cmd:flex
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	cmd:sed
+	cmd:strip
+	cmd:tar
+	cmd:xargs
+	"
+
+
+BUILD_PACKAGE_ACTIVATION_PHASE=INSTALL
+
+sourceDir=$(pwd)
+relativeInstallDir="develop/tools$secondaryArchSubDir"
+installDir="$prefix/$relativeInstallDir"
+objectsDir=$(pwd)/../${portVersionedName}-obj
+
```

Comment:
For the tests:
```diff
+TEST_REQUIRES="
+	cmd:runtest
+	"
+
```

Reply:
You could use cmd:runtest, but the system doesn't seem to respond or get a timeout with it, so I wouldn't add it :)
There is one crashing test (test-pexecute) that I removed from the Makefile in as shown here, this completes the tests:
```diff
diff --git a/libiberty/testsuite/Makefile.in b/libiberty/testsuite/Makefile.in
index 6ba0fba..f69a111 100644
--- a/libiberty/testsuite/Makefile.in
+++ b/libiberty/testsuite/Makefile.in
@@ -45,7 +45,7 @@ all:
 check: @CHECK@
 
 really-check: check-cplus-dem check-d-demangle check-rust-demangle \
-		check-pexecute check-expandargv check-strtol
+		check-expandargv check-strtol
 
 # Run some tests of the demangler.
 check-cplus-dem: test-demangle $(srcdir)/demangle-expected
@@ -57,9 +57,6 @@ check-d-demangle: test-demangle $(srcdir)/d-demangle-expected
 check-rust-demangle: test-demangle $(srcdir)/rust-demangle-expected
 	./test-demangle < $(srcdir)/rust-demangle-expected
 
-# Check the pexecute code.
-check-pexecute: test-pexecute
-	./test-pexecute
 
 # Check the expandargv functionality
 check-expandargv: test-expandargv
@@ -78,10 +75,6 @@ test-demangle: $(srcdir)/test-demangle.c ../libiberty.a
 	$(TEST_COMPILE) -o test-demangle \
 		$(srcdir)/test-demangle.c ../libiberty.a
 
-test-pexecute: $(srcdir)/test-pexecute.c ../libiberty.a
-	$(TEST_COMPILE) -DHAVE_CONFIG_H -I.. -o test-pexecute \
-		$(srcdir)/test-pexecute.c ../libiberty.a
-
 test-expandargv: $(srcdir)/test-expandargv.c ../libiberty.a
 	$(TEST_COMPILE) -DHAVE_CONFIG_H -I.. -o test-expandargv \
 		$(srcdir)/test-expandargv.c ../libiberty.a
@@ -101,7 +94,6 @@ install etags tags installcheck:
 # The standard clean rules.
 mostlyclean:
 	rm -f test-demangle
-	rm -f test-pexecute
 	rm -f test-expandargv
 	rm -f test-strtol
 	rm -f demangler-fuzzer
-- 
2.37.3

```

Reply:
I think I prefer to fix the test for the next iteration, rather than patch it out now.